File: apps/android/app/src/main/java/com/questionpro/model/MenuLinks.java
package com.questionpro.model;

import android.graphics.drawable.Drawable;

import com.alibaba.fastjson.JSONObject;

import java.util.Comparator;

/**
 * Created by sachinsable on 22/08/16.
 */

public class MenuLinks implements Comparable<MenuLinks>{
    public int id;
    public Drawable icon;
    public int orderNumber;
    public String name="";
    public String title="";
    public String questionTitle = "";
    public JSONObject data;
    public int badge = 0;
    public int totalResponse = 0;
    public MenuType type;
    public String key="";
    public String url="";
    public String category= "";
    public boolean separator=false;
    public boolean active = false;
    public boolean enabled = true;

    public static MenuLinks fromJSON(JSONObject jsonObject){
        MenuLinks menuLinks = new MenuLinks();
        if(jsonObject.containsKey("name")){
            menuLinks.name = jsonObject.getString("name");
        }
        if(jsonObject.containsKey("title")){
            menuLinks.title = jsonObject.getString("title");
        }
        if(jsonObject.containsKey("type")){
            menuLinks.type = MenuType.valueOf(jsonObject.getString("type"));
        }
        if(jsonObject.containsKey("url")){
            menuLinks.url = jsonObject.getString("url");
        }
        if(jsonObject.containsKey("data")){
            menuLinks.data = jsonObject.getJSONObject("data");
        }
        if(jsonObject.containsKey("separator")){
            menuLinks.separator = jsonObject.getBoolean("separator");
        }
        if(jsonObject.containsKey("orderNumber")){
            menuLinks.orderNumber = jsonObject.getInteger("orderNumber");
        }
        if(jsonObject.containsKey("active")){
            menuLinks.active = jsonObject.getBoolean("active");
        }
        if(jsonObject.containsKey("questionTitle")){
            menuLinks.questionTitle = jsonObject.getString("questionTitle");
        }
        if(jsonObject.containsKey("totalResponse")){
            menuLinks.totalResponse = jsonObject.getInteger("totalResponse");
        }
        if(jsonObject.containsKey("category")){
            menuLinks.category = jsonObject.getString("category");
        }
        return  menuLinks;
    }

    @Override
    public int compareTo(MenuLinks another) {
        return this.orderNumber - another.orderNumber;
    }

    public static JSONObject toJSON(MenuLinks menuLinks){
        JSONObject jsonObject = new JSONObject();
        jsonObject.put("name", menuLinks.name);
        jsonObject.put("title", menuLinks.title);
        jsonObject.put("type", menuLinks.type);
        jsonObject.put("url", menuLinks.url);
        jsonObject.put("data", menuLinks.data);
        jsonObject.put("separator", menuLinks.separator);
        jsonObject.put("eventType","routeChange");
        jsonObject.put("questionTitle",menuLinks.questionTitle);
        jsonObject.put("totalResponse",menuLinks.totalResponse);
        jsonObject.put("category", menuLinks.category);
        return jsonObject;

    }

    public static Comparator<MenuLinks> menuLinksComparator
            = new Comparator<MenuLinks>() {

        public int compare(MenuLinks menu1, MenuLinks menu2) {



            //ascending order
            return menu1.compareTo(menu2);

            //descending order
            //return fruitName2.compareTo(fruitName1);
        }

    };

}

File: apps/ios/ios/GeneralUtilityModule/Model/MenuLinks.swift
//
//  MenuLinks.swift
//  QuestionProCommunities
//
//  Created by Jignesh Raiyani on 8/26/16.
//  Copyright © 2016 QuestionPro Pvt Ltd. All rights reserved.
//

import UIKit

class MenuLinks: NSObject {

    var id : NSNumber?
    var name : String?
    var title : String?
    var data : [String:Any]?
    var url : String?
    var separator : NSNumber?
    var type : String?
    var active : NSNumber?
    var orderNumber : NSNumber?
    var questionTitle : String?
    var totalResponses : NSNumber?
    var category : String?
    
    override init() {
        
    }
    
    class func fromJSON(jsonDict : [String: Any]) -> MenuLinks {
        
        let menuLinks : MenuLinks = MenuLinks()
        
        if let name = jsonDict["name"] as? String {
                menuLinks.name = name
            }
        if let title = jsonDict["title"] as? String {
                menuLinks.title = title
            }
        if let data = jsonDict["data"] as? [String:Any] {
                menuLinks.data = data
            }
        if let type = jsonDict["type"] as? String {
                menuLinks.type = type
            }
        if let url = jsonDict["url"] as? String {
                menuLinks.url = url
            }
        if let separator = jsonDict["separator"] as? Bool {
                menuLinks.separator = NSNumber(value: separator)
            }
        if let active = jsonDict["active"] as? Int {
                menuLinks.active = NSNumber(value: active)
            }
        if let orderNumber = jsonDict["orderNumber"] as? Int {
                menuLinks.orderNumber = NSNumber(value: orderNumber)
            }
        if let questionTitle = jsonDict["questionTitle"] as? String {
                menuLinks.questionTitle = questionTitle
            }
        if let totalResponse = jsonDict["totalResponse"] as? Int {
                menuLinks.totalResponses = NSNumber(value: totalResponse)
            }
        if let category = jsonDict["category"] as? String {
                menuLinks.category = category
            }
        
        return menuLinks
    }
        
}
